### PR TITLE
33 add api for login page

### PR DESCRIPTION
**Pull request by Goulddomo**
Added Register and Login authentication using session IDs. 
Whitelisted paths to restrict access if not authenticated. 
Used Login and Register DTOs

### CODE REVIEW

File: Well-Coffee-UI/src/App.jsx [hunk 1]
```diff
@@ -1,18 +1,111 @@
...
+      {!authenticated ? (
+        <>
+        <Route path="/login" element={<LoginPage setAuthenticated={setAuthenticated} />} />
+        <Route path="/register" element={<Register />} />
+        <Route path="*" element={<Navigate to="/login" />} />
+        </>
+      ) : (
+        <>
+          <Route path="/home" element={<HomePage />} />
+          <Route path="/order" element={<OrderPage />} />
```
Comment (skyoon26) [inline Well-Coffee-UI/src/App.jsx:35]: This needs to be changed to 'element={<OrderFormPage />}' in order to render the correct component, didn't realize the one in pages wasn't being used!

File: Well-Coffee-UI/src/App.jsx [hunk 2]
```diff
@@ -1,18 +1,111 @@
 import NavigationBar from "./components/navigation/NavigationBar";
-import NavigationRoutes from "./components/navigation/NavigationRoutes";
-import Footer from "./components/footer/Footer";
+import { useState } from "react";
+import { Route, 
+  Routes, 
+  Navigate } from "react-router-dom";
+import HomePage from "./components/pages/HomePage";
+import AccountPage from "./components/pages/AccountPage";
+import OrderPage from "./components/pages/OrderPage";
```
Comment (skyoon26) [inline Well-Coffee-UI/src/App.jsx:8]: Also need to reflect the right file: import OrderFormPage from "./components/orderForm/OrderFormPage";

**Comment by skyoon26** [comment]
The changes/comments above should get the 'Make an Order' page to show up! I can fix the footer in another branch.

And I didn't notice at first since I was unknowingly using 'John Doe' and 'password' while testing out the login, but I can't create a new account with a different name and password? The database always shows info for John Doe even if I put in another name and email. I also can't add additional accounts after the first registration. Not sure if that's what should be expected but wanted to point that out!

File: Well-Coffee-UI/src/components/pages/LoginPage.jsx [hunk 3]
```diff
@@ -1,10 +1,43 @@
...
+
+  // const BASEAPIURL = "http://localhost:8080/api/login";
+
+  const handleLogin = async (e) => {
+    e.preventDefault();
+
+    console.log("Is it working?");
+    console.log(email, password);
+    try {
+      const response = await axios.post("http://localhost:8080/api/login",
```
Comment (tgruber9281) [inline Well-Coffee-UI/src/components/pages/LoginPage.jsx:22]: axios call should be made from the appropriate service file

File: Well-Coffee-UI/src/components/pages/LoginPage.jsx [hunk 4]
```diff
@@ -1,10 +1,43 @@
...
+    console.log("Is it working?");
+    console.log(email, password);
+    try {
+      const response = await axios.post("http://localhost:8080/api/login", 
+      {
+        email, 
+        password
+      }, 
+      {
+        withCredentials: true,
```
Comment (tgruber9281) [inline Well-Coffee-UI/src/components/pages/LoginPage.jsx:28]: Does the login call need to be sent with credentials?  The credentials don't exist yet, right?

File: Well-Coffee-UI/src/components/pages/LogoutPage.jsx
Comment (tgruber9281) [inline Well-Coffee-UI/src/components/pages/LogoutPage.jsx:1]: Axios calls should be made in a service file

File: Well-Coffee-UI/src/components/pages/LogoutPage.jsx [hunk 5]
```diff
@@ -1,9 +1,48 @@
...
+    await axios.get("http://localhost:5173/api/logout",
+    { withCredentials: true,
+     });
+     setAuthenticated(false);
+  } catch (error) {
+    console.error("There was an error logging out", error);
+  }
+};
+
+const handleReturn = async () => {
```
Comment (tgruber9281) [inline Well-Coffee-UI/src/components/pages/LogoutPage.jsx:19]: What is this function supposed to do?

File: Well-Coffee-UI/src/components/pages/AccountPage.jsx
Comment (tgruber9281) [inline Well-Coffee-UI/src/components/pages/AccountPage.jsx:1]: On line 10, class should be changed to className

File: Well-Coffee-UI/src/App.jsx (hunk 2 above)
Comment (tgruber9281) [reply Well-Coffee-UI/src/App.jsx:8]: We could also just import the component into the OrderPage file.

File: WellCoffeeInventoryAPI/build.gradle [hunk 6]
```diff
@@ -21,7 +21,14 @@ dependencies {
 	implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
 	implementation 'org.springframework.boot:spring-boot-starter-validation'
 	implementation 'org.springframework.boot:spring-boot-starter-web'
+	implementation 'org.springframework.security:spring-security-crypto'
+	implementation 'io.jsonwebtoken:jjwt-jackson:0.11.5'
```
Comment (tgruber9281) [inline WellCoffeeInventoryAPI/build.gradle:25]: Are we using jwt or jsession?  Unused dependencies should be removed.

File: WellCoffeeInventoryAPI/src/main/java/com/team_lightyear/WellCoffeeInventoryAPI/services/WebApplicationConfig.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.CorsRegistry;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+public class WebApplicationConfig implements WebMvcConfigurer {
+
+//    @Bean
+//    public AuthenticationFilter authenticationFilter() {
```
Comment (tgruber9281) [inline WellCoffeeInventoryAPI/src/main/java/com/team_lightyear/WellCoffeeInventoryAPI/services/WebApplicationConfig.java:13]: Is the AuthenticationFilter.java doing anything since this section is commented out?  Is that file needed since we're not using Thymeleaf?

**Review by tgruber9281 — CHANGES_REQUESTED** [review]

File: Well-Coffee-UI/src/App.jsx (hunk 2 above)
Comment (Goulddomo) [reply Well-Coffee-UI/src/App.jsx:8]: Resolved

File: WellCoffeeInventoryAPI/src/main/java/com/team_lightyear/WellCoffeeInventoryAPI/controllers/LoginController.java [hunk 8]
```diff
@@ -1,22 +1,128 @@
...
+    private static void setAccountInSession(HttpSession session, Account account) {
+        session.setAttribute(accountSessionKey, account.getId());
+    }
+
+    @PostMapping("/login")
+    public ResponseEntity<Map<String, String>> processLoginForm(@RequestBody LoginFormDTO loginFormDTO, HttpServletRequest request) {
+        Map<String, String> responseBody = new HashMap<>();
+        Account theAccount = accountRepository.findByEmail(loginFormDTO.getEmail().toLowerCase());
+        String password = loginFormDTO.getPassword();
+        ResponseEntity response = null;
```
[the remark this answers is not in the document]
Comment (Goulddomo) [reply WellCoffeeInventoryAPI/src/main/java/com/team_lightyear/WellCoffeeInventoryAPI/controllers/LoginController.java:55]: Resolved by adding <Map<String, String> to ResponseEntity

File: WellCoffeeInventoryAPI/src/main/java/com/team_lightyear/WellCoffeeInventoryAPI/dto/RegisterFormDTO.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
+package com.team_lightyear.WellCoffeeInventoryAPI.dto;
+
+public class RegisterFormDTO extends LoginFormDTO {
+    private String firstName;
+
+    private String lastName;
+
+    private String email;
```
[the remark this answers is not in the document]
Comment (Goulddomo) [reply WellCoffeeInventoryAPI/src/main/java/com/team_lightyear/WellCoffeeInventoryAPI/dto/RegisterFormDTO.java:8]: Resolved

File: WellCoffeeInventoryAPI/build.gradle (hunk 6 above)
Comment (Goulddomo) [reply WellCoffeeInventoryAPI/build.gradle:25]: jsession. Resolved

File: Well-Coffee-UI/src/components/pages/LogoutPage.jsx (hunk 5 above)
Comment (Goulddomo) [reply Well-Coffee-UI/src/components/pages/LogoutPage.jsx:19]: Redundant. Resolved

**Comment by gtracey24** [comment]
I just ran the code from the most recent commit and it redirects to localhost:5173/account which can't be found. Other pages like localhost:5173/home work out fine. Also, I can access all the pages without logging in. Sorry if someone already pointed that out!

**Review by tgruber9281 — APPROVED** [review]
Changes look good!
